variables.tf

```diff
@@ -114,5 +133,21 @@ variable "tunneling_enabled" {
 variable "virtual_network_id" {
   type        = string
   default     = null
-  description = "The ID of the virtual network where the Azure Bastion Host is deployed."
+  description = "The ID of the virtual the Developer SKU Bastion hosts is attached to. Required for the Developer SKU Only."
+
+  validation {
+    condition     = (var.sku == "Developer" && var.virtual_network_id != null) || var.sku != "Developer" && var.virtual_network_id == null
+    error_message = "The virtual network ID is required for the Developer SKU (Only)."
+  }
+}
+
+variable "zones" {
+  type        = list(number)

```

Comment:
Should the type be string? Although the language coerces it from number to string, it may be better to use the correct type from the get go. You can see here it is expecting `set(string)`:

- https://github.com/hashicorp/go-azure-helpers/blob/888e4a00388e6a1af161a53f75a181544e1ce91e/resourcemanager/commonschema/zones.go#L57
- https://github.com/hashicorp/terraform-provider-azurerm/blob/e4397b74a9e46660e9832efa1cc89b08706843c5/internal/services/network/bastion_host_resource.go#L166

I appreciate that the public ip module has an incorrect type, but we eventually intend to supply an interface for zones, so better to make it right here and fix public ip later.

Reply:
Done